### DISCUSSION TITLE
$1 bug comes whenever i enter completion

### DISCUSSION THREAD

**Discussion by sofeepokhrel** [joaotavora/eglot#977] [Q&A] [answered]
Hello ,

I'm using eglot with corfu . But i'm having problem in completion. Though completion shows and i can enter and put the completed value. 
But at middle from nowhere it also adds $1
[i asked corfu, they told it is bug of eglot]

For example

This is completing the completion
![image](https://user-images.githubusercontent.com/103311708/174394430-e96a0049-2611-4f28-92ea-7b8b0882d8de.png)

and pressing enter this is result
![image](https://user-images.githubusercontent.com/103311708/174394679-7fe15fd5-50c4-4a5b-9caf-e256a2b68459.png)

This $1 is added into SignupComponent in import.
This not only applies to js but also all other languages and internal completion.

how to solve it?

Thank you.

**Comment by joaotavora** [comment, marked as answer]
> [i asked corfu, they told it is bug of eglot]

I don't know if Corfu is a part of Emacs but Elglot is made to work with Emacs's completion mechanisms, with some light support for the popular "company.el" package.

So without more information about what conditions this problem happens it's impossible.  Maybe ask corfu people to explain to you why they thing Eglot has a bug.

**Reply by sofeepokhrel** [reply]
https://github.com/minad/corfu/issues/194#issuecomment-1159162773

Where he referenced a bug https://elixirforum.com/t/how-is-snippet-expansion-code-completion-supposed-to-work-in-emacs-with-eglot-elixir-ls/19685/5  and told `yassnippets` to solve the problem. but i dont want to install any other package.

**Reply by joaotavora** [reply]
If you don't have yasnippet installed, then I don't think Eglot is communicating to the Language server that it supports snippets.  So if you can show otherwise, then I agree it's a bug.  If you can't it's not.   I find it very hard to believe there's a bug given this code

```lisp
:completion      (list :dynamicRegistration :json-false
                                    :completionItem
                                    `(:snippetSupport
                                      ,(if (eglot--snippet-expansion-fn)
                                           t
                                         :json-false)
                                      :deprecatedSupport t
                                      :tagSupport (:valueSet [1]))
```

and

```lisp
(defun eglot--snippet-expansion-fn ()
  "Compute a function to expand snippets.
Doubles as an indicator of snippet support."
  (and (boundp 'yas-minor-mode)
       (symbol-value 'yas-minor-mode)
       'yas-expand-snippet))
```

... but I have been wrong before.

**Reply by sofeepokhrel** [reply]
I installed yasnippet but i see problem still same  that `$1` is coming in completion.


Do we have import when some completion finishes ?
like when in coding i've `UserService` when i type it completion automatically comes that it matches and as soon as i enter it auto import at top.

How you are config on  eglot with js/ts angular / react / vue ?
Thank you

**Reply by joaotavora** [reply]
> How you are config on eglot with js/ts angular / react / vue ?

Personally, I don't use this.  I've gotten Eglot to work with `typescript-language-server` in the past and I had no problems. You're experiencing one of following:

1. A bug in the language servers you are using
2. A bug in your configuration
3. A bug in Eglot

I only specialize in Eglot, and as far as I can tell, there is no bug.  I don't specialize in your configuration (how could I? I've never seen it!). I also don't specialize in the language servers you are using (again I don't know exactly what they are).  Until you provide this information to someone, along with a clear description of what you are trying to do, I don't know how anyone can help you.
